### PR TITLE
Add a mod that provides XL versions of power armor

### PR DESCRIPTION
**Pull request by ThePotatomancer**
The armor that added has increased encumbrance and special mutant attack allowed.
Can be crafted using a respective power armor and high enough skills in tailoring, 
electronics and mechanics.

### How to use
Get any kind of base game power armor,
get tailoring skill (10 for heavy power armor, 9 for light, 8 for basic and salvaged 7),
get mechanics skill (heavy power armor 5, the rest 4),
get electronics skill (basic power armor and frame 5, the rest 6. salvaged power armor doesn't need electronics),
then craft modified power armor, and enjoy being a mutant spider in power armor.

#### Summary
```SUMMARY: Mods "Added Mutant Power Armor mod"```

#### Purpose of change
Currently, most mutations paths are considered invalid if not a handicap
in the late game, mainly because of locking access to power armor.
Because of that, many players avoid mutation paths that have large body parts to avoid
being locked out of power armor.

#### Describe the solution
By opening an option for late game power armor for non humanoid mutants,
player won't need to hesitate so much about reaching the thresholds for non humanoid 
mutation paths. The extra encumbrance fits with the XL armor theme of many other 
mutant friendly armors, and the recipe only requires somewhat high skill levels to
delay the power armor a bit, as an added balance,

#### Describe alternatives you've considered
I have considered adding a fabrication requirement, but seeing that no actual welding or
crafting is taking place, but only modifications it didn't seem to make much sense.
I have also considered making the recipe more realistic by requiring you to use actual
material to enlarge and modify the armor, but scrapped it since it would require me to first
define how power armor is made, and then rebalance modified power armor volume, value, 
weight and material costs.

### CODE REVIEW

**Comment by roaringjohn** [comment]
You need to lint.

**Comment by Rail-Runner** [comment]
Suggestions below:

> I have considered adding a fabrication requirement, but seeing that no actual welding or
> crafting is taking place, but only modifications it didn't seem to make much sense.

I'd just replace tailoring with fabrication entirely. Tailoring is generally for working with soft materials, and power armor isn't soft.

> I have also considered making the recipe more realistic by requiring you to use actual
> material to enlarge and modify the armor, but scrapped it since it would require me to first
> define how power armor is made, and then rebalance modified power armor volume, value,
> weight and material costs.

Well, if you wanted to add power armor making to core content, it'd need lots of factory equipment to get made (and XL power armor wouldn't exist at all as it requires tight fit, though *maybe* it could be modified to fit a specific mutations set with same factory tools requirement). Though since it's a mod, it doesn't have to represent what's actually possible.

For tools, I'd add a welding requirement, a soldering requirement, metal cutting quality, bolt turning quality, screw driving quality, glare protection quality and hammering quality. For components, I'd add a second identical power armor (to cannibalize for parts).

**Comment by ThePotatomancer** [comment]
I have passed the entire mod through the [web formatter](http://dev.narc.ro/cataclysm/format.html)
and I still cannot pass the lint tests, and I can't figure out the problem. 
Could someone point me what I am doing wrong?
Do I have to build a local formatter?

**Comment by AMurkin** [comment]
Add one newline at the end.
pic

File: data/mods/Mutant_Power_Armor/armor.json [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    "encumbrance": 50,
+    "storage": 16,
+    "warmth": 60,
+    "power_armor": true,
+    "material_thickness": 9,
+    "environmental_protection": 16,
+    "use_action": { "type": "ups_based_armor", "activate_msg": "Your power armor engages." },
+    "flags": [ "WATERPROOF", "STURDY", "ELECTRIC_IMMUNE", "OVERSIZE", "ALLOWS_NATURAL_ATTACKS" ]
+  }
+]
```
Comment (anothersimulacrum) [inline data/mods/Mutant_Power_Armor/armor.json]:
```suggestion
]

```

File: data/mods/Mutant_Power_Armor/modinfo.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    "type": "MOD_INFO",
+    "ident": "Mutant_Power_Armor",
+    "name": "Mutant Power Armor",
+    "authors": [ "Potatomancer" ],
+    "maintainers": [ "Potatomancer" ],
+    "description": "Adds more encumbaring versions of power armor for oversized mutants. Now more mutants are end game viable!",
+    "category": "items",
+    "dependencies": [ "dda" ]
+  }
+]
```
Comment (anothersimulacrum) [inline data/mods/Mutant_Power_Armor/modinfo.json]:
```suggestion
]

```

File: data/mods/Mutant_Power_Armor/recipes/armor.json [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      { "id": "SAW_M", "level": 2 },
+      { "id": "WRENCH", "level": 1 },
+      { "id": "HAMMER_FINE", "level": 1 },
+      { "id": "SCREW_FINE", "level": 1 },
+      { "id": "SAW_M_FINE", "level": 1 }
+    ],
+    "byproducts": [ [ "alloy_plate", 2 ] ],
+    "components": [ [ [ "power_armor_light", 2 ] ] ]
+  }
+]
```
Comment (anothersimulacrum) [inline data/mods/Mutant_Power_Armor/recipes/armor.json]:
```suggestion
]

```

File: data/mods/Mutant_Power_Armor/armor.json [hunk 4]
```diff
@@ -0,0 +1,216 @@
+[
+  {
+    "id": "depowered_mutant_armor",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "salvaged modified power armor",
+    "description": "The DoubleTech Power Armor, Mk. I: A heavy suit of basic power armor, offering very good protection against attacks, but hard to move in.  This suit has had its servos and cooling system stripped out, meaning it no longer requires power, but also encumbers you greatly and doesn't provide internal thermal regulation.",
+    "weight": 12214,
+    "volume": 100,
```
Comment (anothersimulacrum) [inline data/mods/Mutant_Power_Armor/armor.json]:
```suggestion
    "volume": "25 L",
```

File: data/mods/Mutant_Power_Armor/armor.json [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    "flags": [ "WATERPROOF", "STURDY", "OVERSIZE", "ALLOWS_NATURAL_ATTACKS" ]
+  },
+  {
+    "id": "depowered_mutant_helmet",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "salvaged modified power armor helmet",
+    "description": "A basic helmet, designed for use with the DoubleTech Power Armor, Mk. I.  Offers excellent protection from both attacks and environmental hazards.  This suit has had its internal computer and cooling system stripped out, meaning it no longer requires power, but it has no internal chronometer and doesn't provide internal thermal regulation.",
+    "weight": 2416,
+    "volume": 20,
```
Comment (anothersimulacrum) [inline data/mods/Mutant_Power_Armor/armor.json]:
```suggestion
    "volume": "5 L",
```

File: data/mods/Mutant_Power_Armor/armor.json [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    "flags": [ "WATERPROOF", "STURDY", "SUN_GLASSES", "OVERSIZE", "ALLOWS_NATURAL_ATTACKS" ]
+  },
+  {
+    "id": "mutant_power_armor_basic",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "modified basic power armor",
+    "description": "The DoubleTech Power Armor, Mk. I: A heavy suit of basic power armor, offering very good protection against attacks, but hard to move in.  The UPS compartment can store other things if you don't mind powering the suit yourself.  Like all DoubleTech power armor, the control and climate-regulation systems require direct skin contact.",
+    "weight": 20339,
+    "volume": 100,
```
Comment (anothersimulacrum) [inline data/mods/Mutant_Power_Armor/armor.json]:
```suggestion
    "volume": "25 L",
```

File: data/mods/Mutant_Power_Armor/armor.json [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+    "flags": [ "WATERPROOF", "STURDY", "ELECTRIC_IMMUNE", "OVERSIZE", "ALLOWS_NATURAL_ATTACKS" ]
+  },
+  {
+    "id": "mutant_power_armor_frame",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": "modified power armor hauling frame",
+    "description": "A heavy duty hauling frame designed to interface with power armor.",
+    "weight": 1640,
+    "volume": 48,
```
Comment (anothersimulacrum) [inline data/mods/Mutant_Power_Armor/armor.json]:
```suggestion
    "volume": "12 L",
```

Comment (ThePotatomancer) [reply data/mods/Mutant_Power_Armor/armor.json]: thanks a bunch! I was honestly stumped by the end of file linebreak

File: data/mods/Mutant_Power_Armor/modinfo.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "type": "MOD_INFO",
+    "ident": "Mutant_Power_Armor",
+    "name": "Mutant Power Armor",
+    "authors": [ "ThePotatomancer" ],
+    "description": "Adds more encumbaring versions of power armor for oversized mutants. Now more mutants are end game viable!",
```
Comment (Night-Pryanik) [inline data/mods/Mutant_Power_Armor/modinfo.json]:
```suggestion
    "description": "Adds more encumbering versions of power armor for oversized mutants. Now more mutants are end game viable!",
```

**Comment by Night-Pryanik** [comment]
Oh no, another content-mod is going to be in the official bundle. Don't like this. I think nexusmods Cataclysm page is far more fitting for mods like this.

**Comment by ThePotatomancer** [comment]
umm I don't think that gorgon build should fail because of a pure typo fix.
is there a way to rerun the build without commiting?

**Comment by anothersimulacrum** [comment]
Jenkins rebuild

**Comment by LaVeyanFiend** [comment]
> umm I don't think that gorgon build should fail because of a pure typo fix.
is there a way to rerun the build without commiting?

Its failure isn't your fault, the commiters know it's being unreliable. Don't worry about it for now.

**Comment by anothersimulacrum** [comment]
> is there a way to rerun the build without commiting?
Sorry, I thought I had edited that post to tell you how it worked.
You can comment `Jenkins rebuild` to get the gorgon build to run again.

**Comment by ThePotatomancer** [comment]
Jenkins rebuild

**Comment by ThePotatomancer** [comment]
Jenkins rebuild

**Comment by Petethegoat** [comment]
I agree that we shouldn't necessarily add more mods to the repo, but is there any reason this shouldn't be mainlined? Recipe requirements sound reasonable to me.

**Comment by kevingranade** [comment]
If you mean adding this content to the base game, that's out of the question, construction of power armor is explicitly outside the scope of what survivors can do.

**Pull request closed without merging**